src/routers/StackRouter.js

```diff
@@ -348,6 +303,60 @@ export default (routeConfigs, stackConfig = {}) => {
         }
       }
 
+      // Handle pop-to-top behavior. Make sure this happens after children have had a chance to handle the action, so that the inner stack pops to top first.
+      if (action.type === NavigationActions.POP_TO_TOP) {
+        if (state.index === 0) {
+          return {
+            ...state,
```

Comment:
this looks crazy, without an explanation of why its happening

Reply:
this actually didn't change but I can add a comment here